src/pages/TestK8s.tsx

```diff
@@ -0,0 +1,231 @@
+          // Lock in the namespace
+          let ns = null;
+          if (Array.isArray(data)) {
+            const namespaces = data.map((dataResource) => dataResource.metadata.name);
+            console.debug('++++available namespaces:', namespaces);
+            ns = namespaces[0];
+          } else if (data?.metadata?.namespace) {
+            ns = data.metadata.namespace;
+          }
+
+          if (ns) {
+            setAction(null); // prevent re-invoking this effect/call
+            setNamespace(ns);
+          } else {
+            // eslint-disable-next-line no-alert
+            alert(
+              'Could not find namespace; you are likely not able to do much as we are targeting "default"',
+            );
+          }
+          return data;
+        });
+        break;
+      case ActionType.CREATEAPP:
+        promise = k8sCreateResource({
+          model: ApplicationModel,
+          data: applicationData,
+        });
+        break;
+      case ActionType.GETAPP:
+        promise = k8sGetResource({
+          model: ApplicationModel,
+          name: applicationData.metadata.name,
+          ns: applicationData.metadata.namespace,
+        });
+        break;
+      case ActionType.CREATECOMP:
+        promise = k8sCreateResource({
+          model: ComponentModel,
+          data: componentData,
+        });
+        break;
+      case ActionType.GETCOMP:
+        promise = k8sGetResource({
+          model: ComponentModel,
+          name: componentData.metadata.name,
+          ns: applicationData.metadata.namespace,
+        });
+        break;
+      case ActionType.PATCH:
+        promise = k8sPatchResource({
+          model: ApplicationModel,
+          resource: applicationData,
+          data: [
+            {
+              op: 'replace',
+              path: '/spec/displayName',
+              value: 'New Test Application',
+            },
+          ],
+        });
+        break;
+      case ActionType.PUT:
+        promise = k8sUpdateResource({
+          model: ConfigMapModel,
+          name: testConfigMapMetadata.metadata.name,
+          ns: testConfigMapMetadata.metadata.namespace,
+          data: { ...testConfigMapData, data: { ...testConfigMapData.data, new: 'prop' } },
+        });
+        break;
+      case ActionType.DELETE:
+        promise = k8sDeleteResource({
+          model: ComponentModel,
+          resource: componentData,
+        });
+        break;
+      case null:
+        // ignore effect
+        break;
+      default:
+        // this shouldn't happen, catch state for missed cases
+        throw new Error('uh oh!');
+    }
+    promise
+      ?.then((data) => {
+        setStatus(`${action} response:`);
+        setR(data);
+        console.debug(`++++${action}!`, data);
+      })
+      .catch((err) => {
+        console.error(`++++failed ${action}`, err);
+        setStatus('failed call');
+        setR(null);
+      });
+  }, [action, name, namespace]);
```

Comment:
Not a big deal since this is just a test page, but this effect is firing on every keystroke. Is it okay for it to be that way? Maybe we can't do much about it

Reply:
Yeah, we don't need to worry about this right now. Its just for testing APIs.